[0.6.0] Invalid memory address or nil pointer dereference on diff and sync

Hi, I've recently upgraded to version 0.6.0 from 0.5.2. I've successfully dumped workspace yaml files with the --with-id flag 

> deck dump --all-workspaces --with-id --kong-addr=value

and tried to execute a diff command with one of the dumped files on my local instance 

> deck diff -s default.yaml

but the following error appears:

``` 
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x14fefec]

goroutine 5 [running]:
github.com/hbagdi/deck/cmd.checkWorkspace(0x162c976, 0x15, 0xc0003cc498, 0x7, 0x1ababd8, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
        /home/hbagdi/deck/cmd/common.go:45 +0x18c
github.com/hbagdi/deck/cmd.syncMain(0x7ffeefbff9b5, 0x1a, 0xc0001cbd01, 0xa, 0x0, 0x0)
        /home/hbagdi/deck/cmd/common.go:64 +0xbf
github.com/hbagdi/deck/cmd.glob..func1(0x1a95460, 0xc00000c0c0, 0x0, 0x2, 0x0, 0x0)
        /home/hbagdi/deck/cmd/diff.go:26 +0x4a
github.com/spf13/cobra.(*Command).execute(0x1a95460, 0xc00000c080, 0x2, 0x2, 0x1a95460, 0xc00000c080)
        /home/hbagdi/go/pkg/mod/github.com/spf13/cobra@v0.0.3/command.go:762 +0x460
github.com/spf13/cobra.(*Command).ExecuteC(0x1a94ae0, 0x0, 0x0, 0x0)
        /home/hbagdi/go/pkg/mod/github.com/spf13/cobra@v0.0.3/command.go:852 +0x2ea
github.com/spf13/cobra.(*Command).Execute(...)
        /home/hbagdi/go/pkg/mod/github.com/spf13/cobra@v0.0.3/command.go:800
github.com/hbagdi/deck/cmd.Execute.func2(0xc00005e010, 0xc000022070)
        /home/hbagdi/deck/cmd/root.go:59 +0x2d
created by github.com/hbagdi/deck/cmd.Execute
        /home/hbagdi/deck/cmd/root.go:58 +0xba
```

How can I resolve this?

--- a/cmd/common.go
+++ b/cmd/common.go
@@ -36,7 +36,7 @@ func checkWorkspace(config utils.KongClientConfig) error {
 		return err
 	}
 
-	req, err := http.NewRequest("GET", config.Address+"/workspace/"+workspace,
+	req, err := http.NewRequest("GET", config.Address+"/workspaces/"+workspace,
 		nil)
 	if err != nil {
 		return err

--- a/cmd/common.go
+++ b/cmd/common.go
@@ -42,12 +42,16 @@ func checkWorkspace(config utils.KongClientConfig) error {
 		return err
 	}
 	resp, err := client.Do(nil, req, nil)
+	if err != nil {
+		return errors.Wrapf(err, "checking workspace '%v' in Kong", workspace)
+	}
 	if resp.StatusCode == 404 {
 		return errors.Errorf("workspace '%v' does not exist in Kong, "+
 			"please create it before running decK.", workspace)
 	}
-	if err != nil {
-		return errors.Wrapf(err, "checking workspace '%v' in Kong", workspace)
+	if resp.StatusCode != 200 {
+		return errors.Errorf("unexpected error code while retrieving "+
+			"workspace '%v' : %v", workspace, resp.StatusCode)
 	}
 	return nil
 }

--- a/file/builder.go
+++ b/file/builder.go
@@ -421,6 +421,7 @@ func (b *stateBuilder) upstreams() {
 	}
 
 	for _, u := range b.targetContent.Upstreams {
+		u := u
 		if utils.Empty(u.ID) {
 			ups, err := b.currentState.Upstreams.Get(*u.Name)
 			if err == state.ErrNotFound {

--- a/file/builder_test.go
+++ b/file/builder_test.go
@@ -1294,6 +1294,116 @@ func Test_stateBuilder_upstream(t *testing.T) {
 				},
 			},
 		},
+		{
+			name: "multiple upstreams are handled correctly",
+			fields: fields{
+				targetContent: &Content{
+					Upstreams: []FUpstream{
+						{
+							Upstream: kong.Upstream{
+								Name: kong.String("foo"),
+							},
+						},
+						{
+							Upstream: kong.Upstream{
+								Name: kong.String("bar"),
+							},
+						},
+					},
+				},
+				currentState: emptyState(),
+			},
+			want: &utils.KongRawState{
+				Upstreams: []*kong.Upstream{
+					{
+						ID:    kong.String("5b1484f2-5209-49d9-b43e-92ba09dd9d52"),
+						Name:  kong.String("foo"),
+						Slots: kong.Int(10000),
+						Healthchecks: &kong.Healthcheck{
+							Active: &kong.ActiveHealthcheck{
+								Concurrency: kong.Int(10),
+								Healthy: &kong.Healthy{
+									HTTPStatuses: []int{200, 302},
+									Interval:     kong.Int(0),
+									Successes:    kong.Int(0),
+								},
+								HTTPPath:               kong.String("/"),
+								HTTPSVerifyCertificate: kong.Bool(true),
+								Type:                   kong.String("http"),
+								Timeout:                kong.Int(1),
+								Unhealthy: &kong.Unhealthy{
+									HTTPFailures: kong.Int(0),
+									TCPFailures:  kong.Int(0),
+									Timeouts:     kong.Int(0),
+									Interval:     kong.Int(0),
+									HTTPStatuses: []int{429, 404, 500, 501, 502, 503, 504, 505},
+								},
+							},
+							Passive: &kong.PassiveHealthcheck{
+								Healthy: &kong.Healthy{
+									HTTPStatuses: []int{200, 201, 202, 203, 204, 205,
+										206, 207, 208, 226, 300, 301, 302, 303, 304, 305,
+										306, 307, 308},
+									Successes: kong.Int(0),
+								},
+								Unhealthy: &kong.Unhealthy{
+									HTTPFailures: kong.Int(0),
+									TCPFailures:  kong.Int(0),
+									Timeouts:     kong.Int(0),
+									HTTPStatuses: []int{429, 500, 503},
+								},
+							},
+						},
+						HashOn:           kong.String("none"),
+						HashFallback:     kong.String("none"),
+						HashOnCookiePath: kong.String("/"),
+					},
+					{
+						ID:    kong.String("dfd79b4d-7642-4b61-ba0c-9f9f0d3ba55b"),
+						Name:  kong.String("bar"),
+						Slots: kong.Int(10000),
+						Healthchecks: &kong.Healthcheck{
+							Active: &kong.ActiveHealthcheck{
+								Concurrency: kong.Int(10),
+								Healthy: &kong.Healthy{
+									HTTPStatuses: []int{200, 302},
+									Interval:     kong.Int(0),
+									Successes:    kong.Int(0),
+								},
+								HTTPPath:               kong.String("/"),
+								HTTPSVerifyCertificate: kong.Bool(true),
+								Type:                   kong.String("http"),
+								Timeout:                kong.Int(1),
+								Unhealthy: &kong.Unhealthy{
+									HTTPFailures: kong.Int(0),
+									TCPFailures:  kong.Int(0),
+									Timeouts:     kong.Int(0),
+									Interval:     kong.Int(0),
+									HTTPStatuses: []int{429, 404, 500, 501, 502, 503, 504, 505},
+								},
+							},
+							Passive: &kong.PassiveHealthcheck{
+								Healthy: &kong.Healthy{
+									HTTPStatuses: []int{200, 201, 202, 203, 204, 205,
+										206, 207, 208, 226, 300, 301, 302, 303, 304, 305,
+										306, 307, 308},
+									Successes: kong.Int(0),
+								},
+								Unhealthy: &kong.Unhealthy{
+									HTTPFailures: kong.Int(0),
+									TCPFailures:  kong.Int(0),
+									Timeouts:     kong.Int(0),
+									HTTPStatuses: []int{429, 500, 503},
+								},
+							},
+						},
+						HashOn:           kong.String("none"),
+						HashFallback:     kong.String("none"),
+						HashOnCookiePath: kong.String("/"),
+					},
+				},
+			},
+		},
 	}
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {

